metrics/core/src/main/java/org/hyperledger/besu/metrics/ObservableMetricsSystem.java

```diff
@@ -17,11 +17,33 @@
 import org.hyperledger.besu.plugin.services.MetricsSystem;
 import org.hyperledger.besu.plugin.services.metrics.MetricCategory;
 
+import java.util.Set;
 import java.util.stream.Stream;
 
 public interface ObservableMetricsSystem extends MetricsSystem {
 
   Stream<Observation> streamObservations(MetricCategory category);
 
   Stream<Observation> streamObservations();
+
+  /**
+   * Provides an immutable view into the metric categories enabled for metric collection.
+   *
+   * @return the set of enabled metric categories.
+   */
+  Set<MetricCategory> getEnabledCategories();
+
+  /**
+   * Checks if a particular category of metrics is enabled.
+   *
+   * @param category the category to check
+   * @return true if the category is enabled, false otherwise
+   */
+  default boolean isCategoryEnabled(final MetricCategory category) {
+    return getEnabledCategories().stream()
+        .anyMatch(metricCategory -> metricCategory.getName().equals(category.getName()));
+  }
+
+  /** Stops orderly the metrics system prior to shutdown. */
```

Comment:
remove this in favor of relying on `MetricsService` handling lifecycle events.  The specific sub-classes will know about the specific `MetricsService` they are paired with can can cast down.

Reply:
Done